lib/profile/commands/configure.rb

```diff
@@ -44,26 +50,7 @@ def ask_questions
         prompt.collect do
           type.questions.each do |question|
             key(question.id).ask(question.text) do |q|
-              prefill = type.fetch_answer(question.id)
-              if question.default_smart && prefill.nil?
-                process = Flight::Subprocess::Local.new(
-                  env: {},
-                  logger: smart_log,
-                  timeout: 5,
-                )
-                result = process.run(question.default_smart, nil)
-                output = result.stdout.chomp
-                if !result.success?
-                  smart_log.debug("Command '#{question.default_smart}' failed to run: #{result.stderr}")
-                elsif (!question.validation.has_key?(:format) || output.match(Regexp.new(question.validation.format)))
-                  prefill ||= output
-                else
-                  smart_log.debug("Command result '#{output}' did not pass validation check for '#{question.text}'")
-                end
-              end
-              prefill ||= question.default
-              q.default prefill
-
+              q.default generate_prefill(question)
```

Comment:
`generate_prefill` doesn't exist under the `TTY::Prompt` scope; could you move this to just before the start of the `ask` block please.

Reply:
I thought the functions still worked but instance variables didn't? I could've sworn I tested this and it worked, will check again

Reply:
The first time I ran `configure` it didn't work. I don't see why instance variables and instance methods would behave differently here.

Reply:
You're right - annoyingly it needs to be outside of the entire `prompt.collect` block, which means that having it generate the prompt for the chosen question isn't so simple to do, will have more of a look and do some restructuring